File: java/base/src/main/vsdk/toolkit/media/ShapeDescriptor.java
package vsdk.toolkit.media;
import java.io.Serial;

public abstract class ShapeDescriptor extends MediaEntity
{
    @SuppressWarnings("FieldNameHidesFieldInSuperclass")
    @Serial private static final long serialVersionUID = 20150218L;
    protected String label;

    public ShapeDescriptor(String label)
    {
        setLabel(label);
    }

    public String getLabel()
    {
        return label;
    }

    public final void setLabel(String label)
    {
        this.label = label;
    }

    public double [] getFeatureVector() {
        return null;
    }

    public void setFeatureVector(double vector[]) {

    }
}

File: typescript/base/src/main/vsdk/toolkit/media/ShapeDescriptor.ts
import { MediaEntity } from "./MediaEntity.js";
export abstract class ShapeDescriptor extends MediaEntity {
    protected label: string;
    public constructor(label: string) {
        super();
        this.label = label;
    }
    public getLabel(): string {
        return this.label;
    }
    public setLabel(label: string): void {
        this.label = label;
    }
    public abstract getFeatureVector(): Float64Array;
    public abstract setFeatureVector(vector: Float64Array | number[]): void;
}
